H/HelFEM/build_tarballs.jl

```diff
@@ -0,0 +1,83 @@
+]
+
+script = raw"""
+cp -v ${WORKSPACE}/srcdir/HelFEM/julia/CMake.system ${WORKSPACE}/srcdir/HelFEM/CMake.system
+
+# Set up some platform specific CMake configuration. This is more or less borrowed from
+# the Armadillo build_tarballs.jl script:
+#   https://github.com/JuliaPackaging/Yggdrasil/blob/48d7a89b4aa46b1a8c91269bb138a660f4ee4ece/A/armadillo/build_tarballs.jl#L23-L52
+#
+# We need to manually set up OpenBLAS because FindOpenBLAS.cmake does not work with BB:
+if [[ "${nbits}" == 64 ]] && [[ "${target}" != aarch64* ]]; then
+    OPENBLAS="${libdir}/libopenblas64_.${dlext}"
+else
+    OPENBLAS="${libdir}/libopenblas.${dlext}"
+fi
+
+# Compile libhelfem as a static library
+cd ${WORKSPACE}/srcdir/HelFEM/
+cmake \
+    -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
+    -DUSE_OPENMP=OFF \
+    -DHELFEM_BINARIES=OFF -DHELFEM_FIND_DEPS=ON \
+    -B build/ -S .
+make -C build/ -j${nproc} helfem
+make -C build/ install
+# Copy the HelFEM license
+install_license LICENSE
+
+# Compile the CxxWrap wrapper as a shared library
+cd ${WORKSPACE}/srcdir/HelFEM/julia
+cmake \
+    -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release \
+    -DBLAS_LIBRARIES=${OPENBLAS} \
+    -DJulia_PREFIX=$prefix -DCMAKE_FIND_ROOT_PATH=$prefix -DJlCxx_DIR=$prefix/lib/cmake/JlCxx \
+    -B build/ -S .
+make -C build/ -j${nproc}
+make -C build/ install
+"""
+
+# These are the platforms the libcxxwrap_julia_jll is built on.
+pfkwarg = (; cxxstring_abi = "cxx11")
+platforms = [
+    # x86_64-linux-gnu-cxx11
+    Platform("x86_64", "linux"; libc="glibc", pfkwarg...),
+    # i686-linux-gnu-cxx11
+    Platform("i686", "linux"; libc="glibc", pfkwarg...),
+    # armv7l-linux-gnueabihf-cxx11
+    Platform("armv7l", "linux"; libc="glibc", pfkwarg...),
+    # aarch64-linux-gnu-cxx11
+    Platform("aarch64", "linux"; libc="glibc", pfkwarg...),
+    # x86_64-apple-darwin14-cxx11
+    Platform("x86_64", "macos"; pfkwarg...),
+    # x86_64-w64-mingw32-cxx11
+    Platform("x86_64", "windows"; pfkwarg...),
+    # i686-w64-mingw32-cxx11
+    Platform("i686", "windows"; pfkwarg...),
+    # x86_64-unknown-freebsd11.1-cxx11
+    Platform("x86_64", "freebsd"; pfkwarg...),
+]
+
+products = [
+    LibraryProduct("libhelfem-cxxwrap", :libhelfem),
+]
+
+dependencies = [
+    BuildDependency(PackageSpec(name = "Julia_jll",version = "1.4.1")),
+    Dependency(PackageSpec(name = "libcxxwrap_julia_jll", version = "0.8.0")),
+    Dependency(PackageSpec(name = "armadillo_jll", version = "9.850.1")),
+    Dependency(PackageSpec(name = "GSL_jll", version = "2.6.0")),
+    Dependency(PackageSpec(name = "OpenBLAS_jll", version = "0.3.9")),
+]
+
+build_tarballs(
+    ARGS, name, version, sources, script, platforms, products, dependencies,
+    preferred_gcc_version = v"7.1.0",
```

Comment:
This line is something I am wondering about actually. Without this, I was getting errors about some C++11 features not being enabled because `-std=c++11` is not being passed, which looks like is necessary on GCC 4.x (but not on newer ones). Or something along those lines. I should dig out the exact errors I was getting.

Reply:
The default compiler is GCC v4.8, which I believe doesn't support C++11

Reply:
So the errors that I get appear to be due to "Explicit virtual overrides":

```
[17:08:55] In file included from /workspace/srcdir/HelFEM/libhelfem/src/GaussianNucleus.cpp:16:0:
[17:08:55] /workspace/srcdir/HelFEM/build/libhelfem/include/helfem/GaussianNucleus.h:38:26: warning: override controls (override/final) only available with -std=c++11 or -std=gnu++11
[17:08:55]        double V(double r) const override;
```

They [should be supported since 4.7 actually](https://gcc.gnu.org/projects/cxx-status.html), but seem to require an explicit `-std=c++11` to be passed to the compiler. It might be a CMake configuration issue on my side..

Reply:
Ok, looks like doing `set(CMAKE_CXX_STANDARD 11)` is enough to get this to work without `preferred_gcc_version`. However, it does look like I now need to have `libgfortran_version = v"4"`, since Julia_jll is only built for libgfortran4.